docs/solutions/backup-recovery.md

```diff
@@ -0,0 +1,81 @@
+
+### pgBackRest
+
+[pgBackRest](https://pgbackrest.org/) is an easy-to-use, open-source solution that can reliably back up even the largest of PostgreSQL databases. `pgBackRest` supports the following backup types:
+
+* full backup - a complete copy of your entire dataset.
+* differential backup - includes all data that has changed since the last full backup. While this means the backup time is slightly higher, it enables a faster restore.
+* incremental backup - only backs up the files that have changed since the last full or differential backup, resulting in a quick backup time. To restore to a point in time, however, you will need to restore each incremental backup in the order they were taken.
+
+When it comes to restoring, `pgBackRest` can do a full or a delta restore. A _full_ restore needs an empty PostgreSQL target directory. A _delta_ restore is intelligent enough to recognize already-existing files in the PostgreSQL data directory, and update only the ones the backup contains. 
+
+`pgBackRest` supports remote repository hosting and can even use cloud-based services like AWS S3, Google Cloud Services Cloud Storage, Azure Blob Storage for saving backup files. It supports parallel backup through multi-core processing and compression. By default, backup integrity is verified through checksums, and saved files can be encrypted for enhanced security.
+
+`pgBackRest` can restore a database to a specific point in time in the past. This is the case where a database is not inaccessible but perhaps contains corrupted data. Using the point-in-time recovery, a database administrator can restore the database to the last known good state.  
+
+Finally, `pgBackRest` also supports restoring PostgreSQL databases to a different PostgreSQL instance or a separate data directory.
+
+
+## System architecture
+
+The following diagram illustrates the architecture layout:
+
+![pgBackRest Implementation Architecture](../_images/diagrams/DR-architecture.png)
+
+### Components:
+
+The architecture consists of three server instances:
+
+- `pg-primary` hosts the primary PostgreSQL server. Note that "primary" here means the main database instance and does not refer to the primary node of a PostgreSQL replication cluster or a HA setup.
+- `pg-repo` is the remote backup repository and hosts `pgBackRest`. It’s important to host the backup repository on a physically separate instance, to be accessed when the target goes down. 
+- `pg-secondary` is the _secondary_ PostgreSQL node. Don't confuse it with a hot standby. "Secondary" in this context means a PostgreSQL instance that’s idle. We will restore the database backup to this instance when the primary PostgreSQL instance goes down. 
+
+In our architecture, `pgBackRest` is located on a remote host and the nodes communicate with each other via passwordless SSH.
```

Comment:
We should also add a security warning here that this passwordless SSH configuration may not be ideal for your environment. Please choose one that satisfies your organization's/client's security requirements.

Reply:
Added